frontend/public/components/dashboard/dashboards-page/cluster-dashboard/activity-card.tsx

```diff
@@ -134,11 +154,13 @@ const OngoingActivity = connect(mapStateToProps)(
                 (q) => prometheusResults.getIn([q, 'data']) as PrometheusResponse,
               );
               return {
-                loader: a.properties.loader,
+                loader: (a as DashboardsOverviewPrometheusActivity)?.properties.loader,
+                component: (a as ResolvedExtension<DynamicDashboardsOverviewPrometheusActivity>)
```

Comment:
Mixing both `loader` and `component` causes more code complexity, we should refactor the code to use only one of these, depending on whether we want to defer resolve & render of the given component as async or not.